### DISCUSSION TITLE
Install on Raspi Zero W (32 bit)?

### DISCUSSION THREAD

**Discussion by SR7673** [Kraego/OpenDTU-Grafana-Howto#2] [Q&A] [answered]
Here we are!
I did attempt to install influxdb2 on a Raspi Zero W and found that this tool only works on 64 bit systems. Due to the limited CPU performance and RAM of the zero W i decided to stay at 32 bit and a CLI only (no GUI).
As far as i understand for now, Grafana does not need a GUI-OS because it is configured either via CLI or over a web api, right?
Instead of influxdb2 i would simply try influxdb, which is available. Could that work?
But first of all i need to set up mosquitto...

**Comment by Kraego** [comment, marked as answer]
Hello,
I've also installed it on a headless debian instance over cli. 

And yes grafana doesn't need a UI, you write the settings into a conf file, startup the service and then you open it remote from a browser and finish the setup (add users, create dashboard's, ...). 

Theoretically your setup could work. But I have my doubts regarding the performance of the raspi zero, since the requirements for grafana alone would need almost all the resources (1CPU and 255MB [grafana HW Requirements](https://grafana.com/docs/grafana/latest/setup-grafana/installation/#hardware-recommendations)). So yes but if I had to guess: slow ... but hey never try never know.

Instead of influxdb2 you could use many different DB's as Datasource see [possible datasources](https://grafana.com/docs/grafana/latest/setup-grafana/installation/#hardware-recommendations), with telegraf even more see [telegraf output plugins](https://docs.influxdata.com/telegraf/v1.14/plugins/plugin-list/).

Hope that's helps

**Comment by SR7673** [comment]
Now i'm getting a bit more familiar with MQTT. https://www.elektronik-kompendium.de/sites/raspberry-pi/2709041.htm
Useful. Maybe it is some additional help for others who start with this so i thought it won't hurt to collect some links even here. Usuall i do this on my own but why not doing it public...
